include/simdjson/dom/document-inl.h

```diff
@@ -19,12 +19,17 @@ namespace dom {
 inline element document::root() const noexcept {
   return element(internal::tape_ref(this, 1));
 }
+simdjson_warn_unused
+inline size_t document::capacity() const noexcept {
+  return allocated_capacity;
+}
 
 simdjson_warn_unused
-inline error_code document::allocate(size_t capacity) noexcept {
-  if (capacity == 0) {
+inline error_code document::allocate(size_t len) noexcept {
```

Comment:
I don't mind calling the parameter len, but what are your thoughts leading to the using len instead of capacity? I ask because if we think of a clearer name, I want to replicate it in ondemand :) Not because I want to bikeshed it!

Reply:
Ah. That's because I had the 'capacity' attribute and I did not want a variable clash, but I since renamed the attribute to allocated_capacity so len can be reverted back to capacity.

Done.